test/runtime_test.cc

```diff
@@ -159,12 +175,17 @@ TEST_P(TestVM, Trap) {
   std::string exp_message = "Function: trigger failed";
   ASSERT_TRUE(integration_->error_message_.find(exp_message) != std::string::npos);
 
-  WasmCallWord<1> trigger2;
-  vm_->getFunction("trigger2", &trigger2);
-  EXPECT_TRUE(trigger2 != nullptr);
-  trigger2(current_context_, 0);
-  exp_message = "Function: trigger2 failed:";
-  ASSERT_TRUE(integration_->error_message_.find(exp_message) != std::string::npos);
+  if (runtime_ != "wavm") {
+    // TODO(mathetake): Somehow WAVM exits with 'munmap_chunk(): invalid pointer' on unidentified
```

Comment:
the following is the backtrace from lldb..
```
(lldb) bt all
* thread #1, name = 'runtime_test', stop reason = signal SIGABRT
  * frame #0: 0x00007ffff737f18b libc.so.6`raise + 203
    frame #1: 0x00007ffff735e859 libc.so.6`abort + 299
    frame #2: 0x00007ffff73c93ee libc.so.6`___lldb_unnamed_symbol331$$libc.so.6 + 670
    frame #3: 0x00007ffff73d147c libc.so.6`___lldb_unnamed_symbol399$$libc.so.6 + 28
    frame #4: 0x00007ffff73d16cc libc.so.6`___lldb_unnamed_symbol401$$libc.so.6 + 156
    frame #5: 0x00007ffff75e14c8 libstdc++.so.6`__cxa_demangle + 312
    frame #6: 0x0000555555abc4d3 runtime_test`WAVM::Platform::getInstructionSourceByAddress(unsigned long, WAVM::Platform::InstructionSource&) at DiagnosticsPOSIX.cpp:79:27
    frame #7: 0x000055555596d5b7 runtime_test`WAVM::Runtime::getInstructionSourceByAddress(unsigned long, WAVM::Runtime::InstructionSource&) at Exception.cpp:65:49
    frame #8: 0x000055555596d87a runtime_test`WAVM::Runtime::describeCallStack[abi:cxx11](WAVM::Platform::CallStack const&) at Exception.cpp:106:37
    frame #9: 0x00007ffff7f7ed94 liblibwavm_Ulib.so`proxy_wasm::Wavm::(anonymous namespace)::getFailMessage(function_name=(_M_len = 8, _M_str = "trigger2"), exception=0x00005555599bbce0) at wavm.cc:108:60
    frame #10: 0x00007ffff7f8818d liblibwavm_Ulib.so`void proxy_wasm::getFunctionWavm<proxy_wasm::Word, proxy_wasm::Word>(this=0x00007fffffffd150, exception=0x00005555599bbce0)>*)::'lambda'(proxy_wasm::ContextBase*, proxy_wasm::Word)::operator()(proxy_wasm::ContextBase*, proxy_wasm::Word) const::'lambda0'(WAVM::Runtime::Exception*)::operator()(WAVM::Runtime::Exception*) const at wavm.cc:407:5
 ```

Reply:
this looks really weird because this only happens with `trigger2` function so there's some unhandled tricky corner case in WAVM's `getInstructionSourceByAddress`.

Reply:
what's worse is that I couldn't reproduce this in my working container (Envoy's one) but I could in the bare host Ubuntu..

Reply:
Isn't the VM effectively useless after the first trap? It's not like we're recovering from them, so I don't think this test is representative of what could happen.

Reply:
I thought that but same thing happens even when we have a dedicated separate test for trap2.. but yeah maybe we could delete this trap2 stuff.

Reply:
deleted trigger2.

Reply:
Ah, if it happens in a separate test, then something must be wrong. Perhaps it's because of the return value?

Could you split `trigger2` into a separate test case to make it more representative of a production behavior (untrapped VM getting trapped)? Feel free to skip it for WAVM to unblock this PR, since it's a not a regression. But please add a TODO for it.

Also, could you maybe test this (or similiar) trap in Envoy, where it's hooked end-to-end, to make sure it's not an issue with tests? Thanks!

Reply:
Actually, this test doesn't look right. I don't think the `current_context_` is ever created. Don't you need to add this?
```
TestContext context;
current_context_ = &context;
```

Reply:
added a seprate test for trigger2 and properly set context. But still fails for WAVM in my host ubuntu (and supposedly on GHA as well like before).. Will check on Envoy but not sure I could reproduce. Anyway I skip the test here for WAVM with TODO comment.